api/types/plugin_deviceperm.go

```diff
@@ -0,0 +1,34 @@
+package types
+
+// PluginDevicePermission plugin device permission
+// swagger:model PluginDevicePermission
+// TODO: this is not generated by swagger for some mysterious reason
+type PluginDevicePermission struct {
+
+	// access
+	Access *string `json:"Access,omitempty"`
+
+	// description
+	// Required: true
+	Description string `json:"Description"`
+
+	// major
+	Major *int64 `json:"Major,omitempty"`
+
+	// minor
+	Minor *int64 `json:"Minor,omitempty"`
+
+	// name
+	// Required: true
+	Name string `json:"Name"`
+
+	// settable
+	// Required: true
+	Settable []string `json:"Settable"`
```

Comment:
Do permissions need to be settable as the permissions are added to regular devices anyway. Why does user need to reset permission ranges.